modules/hpe_nimble_volume.py

```diff
@@ -0,0 +1,897 @@
+        required: False
+        type: bool
+        default: True
+        description:
+        - Online state of volume, available for host initiators to establish connections.
+    owned_by_group:
+        required: False
+        type: str
+        description:
+        - Name of group that currently owns the volume.
+    multi_initiator:
+        required: False
+        type: bool
+        default: False
+        description:
+        - For iSCSI volume target, this flag indicates whether the volume and its snapshots can be accessed from multiple initiators at the same time.
+    iscsi_target_scope:
+        required: False
+        type: str
+        choices:
+        - volume
+        - group
+        default: volume
+        description:
+        - This indicates whether volume is exported under iSCSI Group Target or iSCSI volume target. This attribute is only meaningful to iSCSI system.
+    pool:
+        required: False
+        type: str
+        description:
+        - Name associated with the pool in the storage pool table.
+    read_only:
+        required: False
+        type: bool
+        default: False
+        description:
+        - Volume is read-only.
+    block_size:
+        required: False
+        type: int
+        default: 4096
+        description:
+        - Size in bytes of blocks in the volume.
+    clone:
+        required: False
+        type: bool
+        default: False
+        description:
+        - Whether this volume is a clone. Use this attribute in combination with name and snapshot to create a clone by setting clone = true.
+    agent_type:
+        required: False
+        choices:
+        - none
+        - smis
+        - vvol
+        - openstack
+        - openstackv2
+        type: str
+        default: none
+        description:
+        - External management agent type.
+    destination:
+        required: False
+        type: str
+        description:
+        - Name of the destination pool where the volume is moving to.
+    cache_pinned:
+        required: False
+        type: bool
+        default: False
+        description:
+        - If set to true, all the contents of this volume are kept in flash cache.
+    thinly_provisioned:
+        required: False
+        type: bool
+        default: True
+        description:
+        - Set volume's provisioning level to thin.
+    encryption_cipher:
+        required: False
+        choices:
+        - none
+        - aes_256_xts
+        type: str
+        default: none
+        description:
+        - The encryption cipher of the volume.
+    app_uuid:
+        required: False
+        type: str
+        description:
+        - Application identifier of volume.
+    folder:
+        required: False
+        type: str
+        description:
+        - Name of the folder holding this volume.
+    metadata:
+        required: False
+        type: dict
+        description:
+        - Key-value pairs that augment an volume's attributes.
```

Comment:
We need to highlight the fact that these metadata dictionaries are completely arbitrary and user-defined in most cases. The API docs description is also a fair bit longer, not sure how much of that applies to the the Ansible implementation.

Reply:
everything applies to ansible as well. 
how about the below.

- User defined key-value pairs that augment an volume's attributes. List of key-value pairs. Keys must be unique and non-empty. When creating an object, values must be non-empty. When updating an object, an empty value causes the corresponding key to be removed.

OR let me know if you have any description in mind. I am currently going ahead with the one pasted here